src/pybind/mgr/nfs/export.py

```diff
@@ -674,10 +684,6 @@ def create_export_from_dict(self,
                 raise NFSInvalidOperation("export FSAL must specify fs_name")
             if not check_fs(self.mgr, fs_name):
                 raise FSNotFound(fs_name)
-
-            user_id = f"nfs.{cluster_id}.{ex_id}"
-            if "user_id" in fsal and fsal["user_id"] != user_id:
-                raise NFSInvalidOperation(f"export FSAL user_id must be '{user_id}'")
```

Comment:
Oh so now there is no `user_id` validation, i.e. let's say even if someone provided a `user_id = foo` in an/the export block(s), the code would happily accept it, although this has no actual impact since `def _ensure_cephfs_export_user` creates a `user_id` with the cluster name and fs name but this looks wrong to me, I think a proper errno with a err msg must be sent to user that the `user_id` must be the syntax `nfs.<cluster_name>.<fs_name>`, makes sense?

Reply:
Yeah, `user_id` also needs to be generated based on `cmount_path` which will avoid any caps issue. I'll add a commit in this PR itself to fix this

Reply:
Please do check commit message of my last commit, that addresses the issue for generation of `user_id` now

Reply:
So the operators shouldn't/couldn't provide the user id anymore since the code now assigns it in the new func (`fsal.user_id = f"nfs.{export.cluster_id}.{export.fsal.fs_name}"`)?